pagerduty/resource_pagerduty_custom_field_schema_field_configuration.go

```diff
@@ -0,0 +1,257 @@
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = flattenFieldSchemaFieldConfiguration(d, schemaID, createdFieldConfiguration)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func resourcePagerDutyCustomFieldSchemaFieldConfigurationDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	schemaID := d.Get("schema").(string)
+	client, err := meta.(*Config).Client()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	_, err = client.CustomFieldSchemas.DeleteFieldConfigurationContext(ctx, schemaID, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func resourcePagerDutyCustomFieldSchemaFieldConfigurationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := meta.(*Config).Client()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	schemaID, fieldConfiguration, err := buildFieldSchemaFieldConfigurationStruct(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[INFO] Updating PagerDuty field schema field configuration %s", d.Id())
+
+	updatedFieldConfiguration, _, err := client.CustomFieldSchemas.UpdateFieldConfigurationContext(ctx, schemaID, d.Id(), fieldConfiguration)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = flattenFieldSchemaFieldConfiguration(d, schemaID, updatedFieldConfiguration)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func resourcePagerDutyCustomFieldSchemaFieldConfigurationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	schemaID := d.Get("schema").(string)
+	log.Printf("[INFO] Reading PagerDuty field schema field configuration %s %s", schemaID, d.Id())
+	err := fetchFieldSchemaFieldConfiguration(ctx, schemaID, d, meta, handleNotFoundError)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func fetchFieldSchemaFieldConfiguration(ctx context.Context, schemaID string, d *schema.ResourceData, meta interface{}, errorCallback func(error, *schema.ResourceData) error) error {
+	client, err := meta.(*Config).Client()
+	if err != nil {
+		return err
+	}
+
+	return resource.RetryContext(ctx, 2*time.Minute, func() *resource.RetryError {
+		fieldConfiguration, _, err := client.CustomFieldSchemas.GetFieldConfigurationContext(ctx, schemaID, d.Id(), nil)
+		if err != nil {
+			log.Printf("[WARN] Field Schema Field Configuration read error")
+			errResp := errorCallback(err, d)
+			if errResp != nil {
+				time.Sleep(2 * time.Second)
+				return resource.RetryableError(errResp)
+			}
+
+			return nil
+		}
+
+		if err := flattenFieldSchemaFieldConfiguration(d, schemaID, fieldConfiguration); err != nil {
+			return resource.NonRetryableError(err)
+		}
+		return nil
+
+	})
+}
+
+func flattenFieldSchemaFieldConfiguration(d *schema.ResourceData, schemaID string, fc *pagerduty.CustomFieldSchemaFieldConfiguration) error {
+	d.SetId(fc.ID)
```

Comment:
Could you move this `SetId` after the error handling please?

Reply:
That makes sense. Hadn't thought of that.